lib/convection/model/template/resource/aws_s3_bucket.rb

```diff
@@ -30,6 +30,84 @@ def cors_configurationm(*args)
+              config.properties[property.property_name] = opts[property_name] if opts.key?(property_name)
+            end
+
+            config.instance_exec(&block) if block
+            properties['LoggingConfiguration'].set(config)
+          end
+
+          def notification_configuration(&block)
+            config = ResourceProperty::S3NotificationConfiguration.new(self)
+
+            # TODO: Remove this deprecation and remove the opts declaration/usage hash above/below.
+            warn 'DEPRECATED: Defining notification_configuration with an options Hash is deprecated. Please use a configuration block instead. https://github.com/rapid7/convection/pull/143'
+            ResourceProperty::S3NotificationConfiguration.properties.each do |_name, property|
+              config.properties[property.property_name] = opts[property_name] if opts.key?(property_name)
+            end
+
+            config.instance_exec(&block) if block
+            properties['NotificationConfiguration'].set(config)
+          end
+
+          def replication_configuration(&block)
+            config = ResourceProperty::S3ReplicationConfiguration.new(self)
+
+            # TODO: Remove this deprecation and remove the opts declaration/usage hash above/below.
+            warn 'DEPRECATED: Defining replication_configuration with an options Hash is deprecated. Please use a configuration block instead. https://github.com/rapid7/convection/pull/143'
+            ResourceProperty::S3ReplicationConfiguration.properties.each do |_name, property|
+              config.properties[property.property_name] = opts[property_name] if opts.key?(property_name)
+            end
+
+            config.instance_exec(&block) if block
+            properties['ReplicationConfiguration'].set(config)
+          end
+
+          def versioning_configuration(&block)
+            config = ResourceProperty::S3VersioningConfiguration.new(self)
+
+            # TODO: Remove this deprecation and remove the opts declaration/usage hash above/below.
+            warn 'DEPRECATED: Defining versioning_configuration with an options Hash is deprecated. Please use a configuration block instead. https://github.com/rapid7/convection/pull/143'
+            ResourceProperty::S3VersioningConfiguration.properties.each do |_name, property|
+              config.properties[property.property_name] = opts[property_name] if opts.key?(property_name)
+            end
+
+            config.instance_exec(&block) if block
+            properties['VersioningConfiguration'].set(config)
+          end
+
+          def website_configuration(&block)
+            config = ResourceProperty::S3WebsiteConfiguration.new(self)
+
+            # TODO: Remove this deprecation and remove the opts declaration/usage hash above/below.
+            warn 'DEPRECATED: Defining website_configuration with an options Hash is deprecated. Please use a configuration block instead. https://github.com/rapid7/convection/pull/143'
```

Comment:
**No need for a deprecation here!** Looks like `website_configuration` is net new code.